common/types/string_test.go

```diff
@@ -33,6 +36,30 @@ func TestString_Compare(t *testing.T) {
 	}
 }
 
+func TestString_ConvertToNative_Error(t *testing.T) {
+	val, err := String("hello").ConvertToNative(reflect.TypeOf(0))
+	if err == nil {
+		t.Error("Got '%v', expected error", val)
+	}
+}
+
+func TestString_ConvertToNative_Json(t *testing.T) {
+	val, err := String("hello").ConvertToNative(jsonValueType)
+	pbVal := &structpb.Value{&structpb.Value_StringValue{"hello"}}
+	if err != nil ||
+		IsError(val) ||
+		!proto.Equal(val.(proto.Message), pbVal) {
+		t.Error("Error during conversion to json Value type", err, val)
+	}
+}
+
+func TestString_ConvertToNative_String(t *testing.T) {
+	val, err := String("hello").ConvertToNative(reflect.TypeOf(""))
+	if err != nil && val.(string) != "hello" {
```

Comment:
Maybe either of these conditions should be error?

Reply:
Done.